File: backend/GroceryTracker.Api/ProblemTypes.cs
namespace GroceryTracker.Api;

// Stable `type` URIs for ProblemDetails responses. The offline-sync client
// classifies failures (retry / drop / auto-heal) by this string, not by bare
// HTTP status codes, so these values are a contract — don't rename casually.
public static class ProblemTypes
{
    public const string ProfileNotFound = "/errors/profile-not-found";
    public const string TripNotFound = "/errors/trip-not-found";
    public const string ListNotFound = "/errors/list-not-found";
    public const string ItemNotFound = "/errors/item-not-found";
    public const string CategoryNotFound = "/errors/category-not-found";
    public const string CategoryNameConflict = "/errors/category-name-conflict";
    public const string CategoryBuiltIn = "/errors/category-built-in";
    public const string CrossProfile = "/errors/cross-profile";
    public const string Validation = "/errors/validation";
    public const string DbUpdate = "/errors/db-update";
    public const string Unexpected = "/errors/unexpected";
}

File: frontend/src/api/problemTypes.ts
// Mirrors backend/GroceryTracker.Api/ProblemTypes.cs exactly. Kept manually in
// sync (no shared package between the two projects) — the offline outbox's
// failure classifier and plain-language error map key off these strings.
export const ProblemTypes = {
  ProfileNotFound: '/errors/profile-not-found',
  TripNotFound: '/errors/trip-not-found',
  ListNotFound: '/errors/list-not-found',
  ItemNotFound: '/errors/item-not-found',
  CategoryNotFound: '/errors/category-not-found',
  CategoryNameConflict: '/errors/category-name-conflict',
  CategoryBuiltIn: '/errors/category-built-in',
  CrossProfile: '/errors/cross-profile',
  Validation: '/errors/validation',
  DbUpdate: '/errors/db-update',
  Unexpected: '/errors/unexpected',
} as const
